chaoskube/chaoskube.go

```diff
@@ -455,3 +463,32 @@ func filterByOwnerReference(pods []v1.Pod) []v1.Pod {
 
 	return filteredList
 }
+
+// filterByWebhook filters pods by a POST webhook. Only pods where the webhooks returns an
+// HTTP 200 are returned
+func filterByWebhook(pods []v1.Pod, url url.URL) []v1.Pod {
+	// return early if url is not given
+	if url.String() == "" {
+		return pods
+	}
+
+	filteredList := []v1.Pod{}
+
+	for _, pod := range pods {
+		postData := new(bytes.Buffer)
+		err := json.NewEncoder(postData).Encode(pod)
+		if err != nil {
+			continue
+		}
+		resp, err := http.Post(url.String(), "application/json", postData)
+		if err != nil {
+			continue
+		}
+
+		if resp.StatusCode == http.StatusOK {
+			filteredList = append(filteredList, pod)
+		}
+	}
```

Comment:
This will make one request for each candidate pod. If you have a cluster with 1000 pods and none of them aren't filtered out by the previous filters, this will call your endpoint 1000 times.

Is that what you had in mind?

If we want to go with the current behaviour we should document it.

Reply:
Yes that was the idea, I understand that with a large number of pods that might not be workable.. Another option would be to call the webhook only for the randomly selected pod in `Victim()`.

Either way would work for my use case, so maybe it's more wise to move things to `Victim()`? What do you think?